packages/desktop/components/popups/send/SendConfirmationPopup2.svelte

```diff
@@ -0,0 +1,236 @@
+        giftStorageDeposit,
+        visibleSurplus,
+        isInternal,
+        layer2Parameters
+    )
+
+    onMount(async () => {
+        await updateStorageDeposit()
+
+        if (transactionDetails.expirationDate === undefined) {
+            initialExpirationDate = getInitialExpirationDate(expirationDate, storageDeposit, giftStorageDeposit)
+        }
+    })
+
+    async function rebuildTransactionOutput(): Promise<void> {
+        updateNewTransactionDetails({
+            type: transactionType,
+            expirationDate,
+            giftStorageDeposit,
+        })
+
+        try {
+            const transactionDetails = get(newTransactionDetails)
+            const outputParams = await getOutputParameters(transactionDetails)
+            preparedOutput = await prepareOutput($selectedAccount.index, outputParams, getDefaultTransactionOptions())
+
+            await updateStorageDeposit()
+        } catch (err) {
+            handleError(err)
+        }
+    }
+
+    async function updateStorageDeposit(): Promise<void> {
+        const { storageDeposit: _storageDeposit, giftedStorageDeposit: _giftedStorageDeposit } =
+            await getStorageDepositFromOutput($selectedAccount, preparedOutput as CommonOutput)
+
+        storageDeposit = minimumStorageDeposit = _storageDeposit > 0 ? _storageDeposit : _giftedStorageDeposit
+
+        if (isBaseTokenTransfer) {
+            const rawAmount = Number((transactionDetails as NewTokenTransactionDetails).rawAmount)
+            if (rawAmount >= storageDeposit) {
+                storageDeposit = 0
+            }
+        }
+
+        // Note: we need to adjust the surplus
+        // so we make sure that the surplus is always added on top of the minimum storage deposit
+        if (Number(surplus) > 0) {
+            if (minimumStorageDeposit >= Number(surplus)) {
+                visibleSurplus = surplus = undefined
+            } else {
+                visibleSurplus = Number(surplus) - minimumStorageDeposit
+                // Note: we have to hide it because currently, in the sdk,
+                // the storage deposit return strategy is only looked at
+                // if the provided amount is < the minimum required storage deposit
+                hideGiftToggle = true
+            }
+        }
+    }
+
+    function toggleGiftStorageDeposit(): void {
+        giftStorageDeposit = !giftStorageDeposit
+    }
+
+    async function sendOutputAndClosePopup(): Promise<void> {
+        await sendOutput(preparedOutput)
+        closePopup()
+    }
+
+    async function onConfirmClick(): Promise<void> {
+        try {
+            validateSendConfirmation(preparedOutput as CommonOutput)
+
+            if ($isActiveLedgerProfile) {
+                ledgerPreparedOutput.set(preparedOutput)
+            }
+
+            updatePopupProps({ isSendAndClosePopup: true, preparedOutput })
+            await checkActiveProfileAuth(sendOutputAndClosePopup, { stronghold: false, ledger: false })
```

Comment:
Maybe We should leave the `stronghold: true` as We had it before and have popup reopen after returning from stronghold unlock to be in sync with other send flows, also with reopening `sendConfirmationPopup` user gets a feedback that transaction is being sent with Send button `isBusy` spinner. We should than fix issue that expiration  time is not being displayed when returning from unlocking stronghold.